ui/src/selection-panel/selection-panel-geometry-section.tsx

```diff
@@ -0,0 +1,249 @@
+    onToggleSwitchVisibility: (payload: ToggleSwitchPayload) => void;
+    onToggleKmPostVisibility: (payload: ToggleKmPostPayload) => void;
+    togglePlanOpen: (payload: TogglePlanWithSubItemsOpenPayload) => void;
+    openedPlanLayouts: OpenedPlanLayout[];
+    togglePlanKmPostsOpen: (payload: ToggleAccordionOpenPayload) => void;
+    togglePlanAlignmentsOpen: (payload: ToggleAccordionOpenPayload) => void;
+    togglePlanSwitchesOpen: (payload: ToggleAccordionOpenPayload) => void;
+    onSelect: (options: OnSelectOptions) => void;
+};
+const MAX_PLAN_HEADERS = 50;
+
+type LoadedGeometryPlan = {
+    planLayout: GeometryPlanLayout;
+    linkStatus: GeometryPlanLinkStatus;
+};
+
+const SelectionPanelGeometrySection: React.FC<GeometryPlansPanelProps> = ({
+    changeTimes,
+    publishType,
+    selectedItems,
+    viewport,
+    selectedPlanLayouts,
+    trackNumberFilter,
+    onTogglePlanVisibility,
+    onToggleAlignmentVisibility,
+    onToggleSwitchVisibility,
+    onToggleKmPostVisibility,
+    togglePlanOpen,
+    openedPlanLayouts,
+    togglePlanKmPostsOpen,
+    togglePlanAlignmentsOpen,
+    togglePlanSwitchesOpen,
+    onSelect,
+}) => {
+    const { t } = useTranslation();
+    const [planHeadersInView, setPlanHeadersInView] = React.useState<GeometryPlanHeader[]>([]);
+    const [planHeaderCount, setPlanHeaderCount] = React.useState<number>(0);
+    const [loadedPlans, setLoadedPlan] = useMapState<GeometryPlanId, LoadedGeometryPlan>();
+    const [plansBeingLoaded, startLoadingPlan, finishLoadingPlan] = useSetState<GeometryPlanId>();
+
+    React.useEffect(() => {
+        getGeometryPlanHeaders(
+            MAX_PLAN_HEADERS,
+            0,
+            viewport.area,
+            ['GEOVIITE', 'GEOMETRIAPALVELU', 'PAIKANNUSPALVELU'],
+            trackNumberFilter.map((tn) => tn.id),
+            undefined,
+            SortByValue.UPLOADED_AT,
+            SortOrderValue.ASCENDING,
+        ).then((page) => {
+            setPlanHeadersInView(page.items);
+            setPlanHeaderCount(page.totalCount);
+        });
+    }, [viewport.area, changeTimes.geometryPlan, trackNumberFilter]);
+
+    React.useEffect(
+        () => [...loadedPlans.keys()].forEach(loadPlanLayout),
+        [
+            publishType,
+            changeTimes.geometryPlan,
+            changeTimes.layoutReferenceLine,
+            changeTimes.layoutLocationTrack,
+            changeTimes.layoutSwitch,
+            changeTimes.layoutKmPost,
+        ],
+    );
+
+    const loadPlanLayout = (id: GeometryPlanId) => {
+        startLoadingPlan(id);
+        const rv = Promise.all([
+            getTrackLayoutPlan(id, changeTimes.geometryPlan, false),
+            getPlanLinkStatus(id, publishType),
+        ]).then(([planLayout, linkStatus]) => {
+            if (planLayout) {
+                setLoadedPlan(id, { planLayout, linkStatus });
+            }
+            return planLayout;
+        });
+        rv.finally(() => finishLoadingPlan(id));
```

Comment:
Tämän `.finally`:n ei olisi pakko olla omalla rivillään, vaan se voisi olla tuossa rivin 122 perässä. Tällöin sen sijasta että `rv`:n ottaisi muuttujaan, niin sen voisi vaan suoraan returnata jo tuolla rivillä 114.